shared/middleware/files_api.rb

```diff
@@ -430,6 +442,26 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
     }.to_json
   end
 
+  def has_valid_encoding?(source)
+    if source.is_a?(String)
+      return source.force_encoding("UTF-8").valid_encoding?
+    end
+
+    # Handle Java Lab Multi-file Projects
```

Comment:
Nit: To keep ourselves open to future multi-file project types, I wonder if we should should update this language to something like `Handle Multi-file Projects, like Java Lab` or just `Handle Multi-file Projects`.

Reply:
Good point! will update